sklearn/tests/test_naive_bayes.py

```diff
@@ -518,18 +531,16 @@ def test_mnb_prior_unobserved_targets():
 
     clf = MultinomialNB()
 
-    assert_no_warnings(
-        clf.partial_fit, X, y, classes=[0, 1, 2]
-    )
+    with pytest.warns(None):
+        clf.partial_fit(X, y, classes=[0, 1, 2])
```

Comment:
This is not enough to ensure no warning is raised. It just make pytest record the warnings. You need to assert that there are no recorded warning
```suggestion
    with pytest.warns(None) as record:
        clf.partial_fit(X, y, classes=[0, 1, 2])
    assert len(record) == 0
```

Reply:
Thanks for the review @jeremiedbb. Committed the fixes.